velox/expression/SignatureBinder.cpp

```diff
@@ -229,80 +307,53 @@ bool SignatureBinderBase::checkOrSetIntegerParameter(
   return true;
 }
 
-bool SignatureBinderBase::tryBindWithCoercion(
-    const exec::TypeSignature& typeSignature,
-    const TypePtr& actualType,
-    Coercion& coercion) {
-  return tryBind(typeSignature, actualType, true, coercion);
-}
-
-bool SignatureBinderBase::tryBind(
-    const exec::TypeSignature& typeSignature,
-    const TypePtr& actualType) {
-  Coercion coercion;
-  return tryBind(typeSignature, actualType, false, coercion);
-}
-
 bool SignatureBinderBase::tryBind(
     const exec::TypeSignature& typeSignature,
     const TypePtr& actualType,
-    bool allowCoercion,
-    Coercion& coercion) {
-  coercion.reset();
+    Coercion* coercion) {
+  if (!actualType) {
+    return false;
+  }
+
   if (isAny(typeSignature)) {
     return true;
   }
 
   const auto& baseName = typeSignature.baseName();
 
-  if (variables().count(baseName)) {
-    // Variables cannot have further parameters.
+  if (auto varIt = variables().find(baseName); varIt != variables().end()) {
```

Comment:
I assume this is an optimization to avoid double lookup in variables() map. This is nice. Let's move it into a separate PR to keep this PR as simple and short as possible.